ee/codegen/src/context/workflow-output-context.ts

```diff
@@ -23,6 +30,27 @@ export class WorkflowOutputContext {
   }
 
   public getOutputName(): string {
-    return toSnakeCase(this.terminalNodeData.data.name);
+    const name = this.generateUniqueFinalOutputName();
+    return toSnakeCase(name);
+  }
+
+  private generateUniqueFinalOutputName(): string {
+    const value = this.workflowContext.getOutputName(this.terminalNodeData.id);
+    if (value !== undefined) {
+      return value;
+    } else {
+      let counter = 1;
+      const names = new Set(this.workflowContext.outputNamesById.values());
+
+      const originalName = this.terminalNodeData.data.name;
+      let uniqueName = originalName;
+
+      while (names.has(uniqueName)) {
+        uniqueName = `${originalName}-${counter}`;
+        counter++;
+      }
+      this.workflowContext.addOutputName(this.terminalNodeData.id, uniqueName);
```

Comment:
[3/5] I wouldn't expect us to mutate `workflowContext` as part of this context's constructor. aka, We shouldn't call `addOutputName` here.

Instead, we can do this as part of `addWorkflowOutputContext` on the Workflow Context. Basically, as part of registering an output context with a workflow context, we add its output name to the registry. Wdyt?

Instead, it'd be

Reply:
Just updated it